posthog/hogql/ast.py

```diff
@@ -408,6 +408,14 @@ def print_type(self) -> str:
         return "DateTime"
 
 
+@dataclass(kw_only=True)
+class IntervalType(ConstantType):
+    data_type: ConstantDataType = field(default="unknown", init=False)
```

Comment:
`select toTypeName(toIntervalHour(3))` returns `IntervalHour`. Would we benefit from storing the granularity on the type as well?

Reply:
I'm not sure. I think it'd make sense if/when we store interval values beyond just the type information (mainly for the sake of printing them back in their original granularity rather than normalizing them to the lowest common denominator) but otherwise I can't think of a reason it'd make much of a functional difference.